pkg/controllers/cluster-policy-controller.go

```diff
@@ -14,82 +14,84 @@ package controllers
 
 import (
 	"context"
-	"io/ioutil"
-	"os"
+	"fmt"
 	"path/filepath"
 
+	openshiftcontrolplanev1 "github.com/openshift/api/openshiftcontrolplane/v1"
 	clusterpolicycontroller "github.com/openshift/cluster-policy-controller/pkg/cmd/cluster-policy-controller"
-	clusterpolicyversion "github.com/openshift/cluster-policy-controller/pkg/version"
 	"github.com/openshift/library-go/pkg/controller/controllercmd"
 	"github.com/openshift/microshift/pkg/config"
-	"k8s.io/component-base/cli"
+	corev1 "k8s.io/api/core/v1"
+	unstructuredv1 "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
 )
 
 type ClusterPolicyController struct {
-	config     *controllercmd.ControllerCommandConfig
-	flags      *controllercmd.ControllerFlags
+	run        func(context.Context) error
 	kubeconfig string
+
+	configErr error
 }
 
 func NewClusterPolicyController(cfg *config.MicroshiftConfig) *ClusterPolicyController {
 	s := &ClusterPolicyController{}
-	s.configure(cfg)
+	s.configErr = s.configure(cfg)
 	return s
 }
 
 func (s *ClusterPolicyController) Name() string           { return "cluster-policy-controller" }
 func (s *ClusterPolicyController) Dependencies() []string { return []string{"kube-apiserver"} }
 
-func (s *ClusterPolicyController) configure(cfg *config.MicroshiftConfig) {
-
+func (s *ClusterPolicyController) configure(cfg *config.MicroshiftConfig) error {
 	s.kubeconfig = filepath.Join(cfg.DataDir, "resources", "kubeadmin", "kubeconfig")
-	s.writeConfig(cfg)
-
-	// Use NewControllerCommandConfig to create a controller configuration struct
-	// with default values.
-	s.config = controllercmd.NewControllerCommandConfig(s.Name(), clusterpolicyversion.Get(), clusterpolicycontroller.RunClusterPolicyController)
-
-	flags := controllercmd.NewControllerFlags()
-	flags.ConfigFile = filepath.Join(cfg.DataDir, "resources", "cluster-policy-controller", "config", "config.yaml")
-	flags.KubeConfigFile = s.kubeconfig
-	flags.Validate()
-	s.flags = flags
 
+	scheme := runtime.NewScheme()
+	openshiftcontrolplanev1.AddToScheme(scheme)
+	codec := serializer.NewCodecFactory(scheme).LegacyCodec(openshiftcontrolplanev1.GroupVersion)
+
+	encodedConfig, err := runtime.Encode(codec,
+		&openshiftcontrolplanev1.OpenShiftControllerManagerConfig{
+			Controllers: []string{
+				"*",
+				"-openshift.io/resourcequota",
+				"-openshift.io/cluster-quota-reconciliation",
+			},
+		})
+	if err != nil {
+		return err
+	}
+	ctrlConfig := &unstructuredv1.Unstructured{}
```

Comment:
+1 for injecting the config directly without generating a config file

Reply:
+1. @benluddy 's idea :-)